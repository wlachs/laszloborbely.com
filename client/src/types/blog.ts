export type BlogPost = {
	urlHandle: string;
	title: string;
	author: string;
	summary: string;
	body: string;
	creationTime: Date;
};
