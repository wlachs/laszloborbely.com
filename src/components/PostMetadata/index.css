@import '../global.css';

.timestamp {
	white-space: nowrap;
}
