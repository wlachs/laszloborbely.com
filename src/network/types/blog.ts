export type BlogPostData = {
	author: string;
	body: string;
	creationTime: Date;
	id: string;
	summary: string;
	title: string;
};
