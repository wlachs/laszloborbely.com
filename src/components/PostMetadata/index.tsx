import './index.css';

import moment from 'moment/moment';
import { ReactElement } from 'react';

import { type BlogPostData } from '../../network/types/blog';

interface PostMetadataProps {
	post: BlogPostData;
}

export function PostMetadata({ post }: PostMetadataProps): ReactElement {
	const dateFormat = 'LL';
	const timestamp = moment(post.creationTime).format(dateFormat);
	return (
		<div className='pb-2'>
			<h1>{post.title}</h1>
			<span className='timestamp'>{timestamp}</span>
		</div>
	);
}
