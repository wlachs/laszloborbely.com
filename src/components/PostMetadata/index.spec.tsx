import {cleanup, render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {afterEach, describe, expect, it} from 'vitest';
import {type BlogPostData} from '../../network/types/blog.ts';
import PostMetadata from './index.tsx';

const post: BlogPostData = {
	urlHandle: 'test-post',
	title: 'very very long test post',
	author: 'me myself and I',
	summary: 'a summery summary',
	body: 'body with a six-pack of beer',
	creationTime: new Date(0),
};

describe('PostMetadata component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match snapshot', () => {
		const {container} = render(
			<MemoryRouter>
				<PostMetadata post={post}/>,
			</MemoryRouter>,
		);
		expect(container).toMatchSnapshot();
	});
});
