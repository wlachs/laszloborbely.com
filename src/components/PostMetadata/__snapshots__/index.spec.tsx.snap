// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostMetadata component > should match snapshot 1`] = `
<div>
  <div>
    <a
      class=""
      href="/contact"
    >
      me myself and I
    </a>
     â€¢ 
    January 1, 1970
  </div>
  ,
</div>
`;
